### PR TITLE
sd-dhcp6-client: several fixes and cleanups

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd-network/sd-dhcp6-client.c [hunk 1]
```diff
@@ -1309,24 +1294,18 @@ static int client_parse_message(
...
 
         if (client->state != DHCP6_STATE_INFORMATION_REQUEST) {
                 r = dhcp6_lease_get_serverid(lease, NULL, NULL);
-                if (r < 0) {
-                        log_dhcp6_client(client, "%s has no server id",
-                                         dhcp6_message_type_to_string(message->type));
-                        return -EINVAL;
-                }
+                if (r < 0)
+                        return log_dhcp6_client_errno(client, r, "%s has no server id",
```
Comment (bluca) [inline src/libsystemd-network/sd-dhcp6-client.c:1307]: this is changing from always returning `EINVAL` to return `r` - was that intentional?

Comment (yuwata) [reply src/libsystemd-network/sd-dhcp6-client.c:1307]: Yes. Note that the function is called in the following path: `client_receive_message()` -> `client_receive_advertise()/client_receive_reply()` -> `client_parse_message()`, and there is no special handling of `-EINVAL` in that path.

**Review by bluca — APPROVED** [review]
lgtm

**Comment by poettering** [comment]
lgtm, too

**Pull request merged**